src/hugr/views.rs

```diff
@@ -249,12 +299,12 @@ pub trait HugrView: sealed::HugrInternals {
 }
 
 /// A common trait for views of a HUGR hierarchical subgraph.
-pub trait HierarchyView<'a>: HugrView {
-    /// The base from which the subgraph is derived.
-    type Base;
-
+pub trait HierarchyView<'a>: HugrView + Sized {
     /// Create a hierarchical view of a HUGR given a root node.
-    fn new(hugr: &'a Self::Base, root: Node) -> Self;
+    ///
+    /// # Errors
+    /// Returns [`HugrError::InvalidNode`] if the root isn't a node of the required [OpTag]
+    fn new(hugr: &'a impl HugrView, root: Node) -> Result<Self, HugrError>;
```

Comment:
Again, naming questions that I am not the authority on but will still give my opinion -- use it as you wish. I try to prefix function names that can fail with `try_`, so here I would choose `try_new`.

@aborgna-q what is your preference? You're more of a rustacean than me.

Reply:
Yep, this should be [`try_new`](https://doc.rust-lang.org/std/?search=try_new)

Reply:
Sounds good to me, done. I wonder about a utility function `new` that calls `try_new(...).unwrap()` - even if that were `#[cfg(test)]`. I have not done the latter ATM but please shout if you find repeated try_new...unwrap sufficiently annoying to justify this!